src/overlays/actors/ovl_Door_Shutter/z_door_shutter.c

```diff
@@ -21,61 +47,689 @@ const ActorInit Door_Shutter_InitVars = {
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/DoorShutter_Update.s")
+typedef struct {
+    s16 sceneNum;
+    u8 index;
+} ShutterSceneInfo;
+
+static ShutterSceneInfo D_80998240[] = {
+    { SCENE_YDAN, 0x02 },       { SCENE_DDAN, 0x03 },         { SCENE_DDAN_BOSS, 0x03 },
+    { SCENE_BDAN, 0x04 },       { SCENE_BMORI1, 0x05 },       { SCENE_HIDAN, 0x08 },
+    { SCENE_GANON, 0x09 },      { SCENE_GANON_BOSS, 0x09 },   { SCENE_JYASINZOU, 0x0A },
+    { SCENE_JYASINBOSS, 0x0A }, { SCENE_MIZUSIN, 0x0B },      { SCENE_HAKADAN, 0x0C },
+    { SCENE_HAKADANCH, 0x0C },  { SCENE_ICE_DOUKUTO, 0x0D },  { SCENE_MEN, 0x0E },
+    { SCENE_GANONTIKA, 0x0F },  { SCENE_HAKAANA_OUKE, 0x10 }, { -1, 0x07 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80997838.s")
+typedef struct {
+    s16 dungeonScene;
+    s16 bossScene;
+    u8 index;
+} BossDoorInfo;
+
+static BossDoorInfo D_80998288[] = {
+    { SCENE_HIDAN, SCENE_FIRE_BS, 0x01 },
+    { SCENE_MIZUSIN, SCENE_MIZUSIN_BS, 0x02 },
+    { SCENE_HAKADAN, SCENE_HAKADAN_BS, 0x03 },
+    { SCENE_GANON, SCENE_GANON_BOSS, 0x04 },
+    { SCENE_BMORI1, SCENE_MORIBOSSROOM, 0x05 },
+    { SCENE_JYASINZOU, SCENE_JYASINBOSS, 0x06 },
+    { -1, -1, 0x00 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80997A34.s")
+static Gfx* D_809982B4[] = {
+    0x06000590, 0x06000BF0, 0x06002BD0, 0x060018B0, 0x06001F10, 0x060018B0, 0x06001250, 0x06000BF0,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/DoorShutter_Draw.s")
+// boss door textures?
+static UNK_PTR D_809982D4[] = {
+    0x060065C0, 0x060035C0, 0x060055C0, 0x060045C0, 0x06000000, 0x060025C0, 0x060015C0,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_8099803C.s")
+extern ColHeader D_0601EDD0; // gohma block collision header
+extern ColHeader D_06012FD0; // phantom ganon bars collision header
+
+void DoorShutter_SetupAction(DoorShutter* this, DoorShutterActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+    this->unk_16F = 0;
+}
+
+s32 DoorShutter_SetupDoor(DoorShutter* this, GlobalContext* globalCtx) {
+    TransitionActorEntry* transitionEntry = &globalCtx->transitionActorList[(u16)this->dyna.actor.params >> 0xA];
+    s8 frontRoom = transitionEntry->sides[0].room;
+    s32 doorType = this->doorType;
+    ShutterObjectInfo* temp_t0 = &D_809980F0[this->unk_16B];
+
+    if (doorType != SHUTTER_KEY_LOCKED) {
+        // if front room equals back room (no room transition)
+        if (frontRoom == transitionEntry->sides[1].room) {
+            if (ABS((s16)(this->dyna.actor.shape.rot.y - this->dyna.actor.yawTowardsLink)) < 0x4000) {
+                frontRoom = 0xFF;
+            }
+        }
+        if (frontRoom == this->dyna.actor.room) {
+            if (doorType == SHUTTER_FRONT_SWITCH_BACK_CLEAR) { // Swap the back clear to the front clear
+                doorType = SHUTTER_FRONT_CLEAR;
```

Comment:
Considering the current FRONT_SWITCH/CLEAR types check switch and clear flags respectively I believe I have them the right way around.
The names were inspired by cloudmodding actor params documentation and then ensuring it made sense in the code, which it does for the majority of cases. Skimming over the usages of `doorType` again, this highlighted case seems to be the only oddity as far as I can tell?
I can't say I fully understand what is going on there either at this time but this file needs further documentation anyway which I am not interested in doing myself at least not at this time.

Reply:
I guess where I'm really confused about is `frontRoom == this->dyna.actor.room` and what exactly does `this->dyna.actor.room` mean for a door

what do types like `SHUTTER_FRONT_CLEAR` do when the door is opened from the back?

Reply:
Types with an unspecified `BACK` behave like an unlocked/unbarred door from the other side. I think the concepts of "front" and "back" for these doors may be misleading since the "front" of the door may not always be in the direction you're coming from.